src/master-playlist-controller.js

```diff
@@ -1132,6 +1135,8 @@ export class MasterPlaylistController extends videojs.EventTarget {
       return;
     }
 
+    currentPlaylist.retries++;
```

Comment:
I think this value may not reflect exactly what we mean. Right now, if we try to request a playlist once and it fails, we will set `retries` to 1. Technically it wasn't a retry, it was just the first attempt. We may actually never retry the playlist if the selection algorithm chooses a different one later on.

In addition, this doesn't take into consideration the blacklist reason, meaning `retries` will be incremented for some cases where we haven't actually retried a playlist (e.g., excessive segment downloads, an audio track failed to load, and early aborts)

Reply:
> I think this value may not reflect exactly what we mean. Right now, if we try to request a playlist once and it fails, we will set `retries` to 1. Technically it wasn't a retry, it was just the first attempt. We may actually never retry the playlist if the selection algorithm chooses a different one later on.

I think that is fine for this pass, as this retry is playlist dependent. A playlist should be excluded if we have retried it a couple of times, it doesn't matter whether we switched away or not. The name may not be ideal, but I don't think it should stop us from getting this in.

> In addition, this doesn't take into consideration the blacklist reason, meaning `retries` will be incremented for some cases where we haven't actually retried a playlist (e.g., excessive segment downloads, an audio track failed to load, and early aborts)

That's a good point. Are those things we have reasons available for in this method?

Reply:
For some we do, for others we don't. I think we may be best off determining exactly what type of reason(s) we'd want to count for this option, ensure we are passing that reason from callers of `blacklistCurrentPlaylist`, and while we're at it maybe update the `retries` property to something more precise or use `retries` but change our logic around it, since the playlist object is public and we are not marking this property as private. Alternatively, we can mark it as private.

Reply:
So, the places we call `blacklistCurrentPlaylist` are:
- playback watcher for excessive download (Infinity)
- media groups, error encountered with an audio group
- MPC, on a playlist loader error
- MPC, on playlistunchanged if we determine it's an outdated playlist
- MPC, segment loader error
- MPC, earlyabort (this one is going away)
- MPC, unknown codecs (Infinity)
- MPC, unsupported codecs (Infinity)
- MPC, codec switching not supported (Infinity)

While the name `retires` may not necessarily apply to all those conditions, I think it's reasonable to increment on all of these. Maybe it's unnecessary to increment if we are excluding playlists indefinitely, but I don't think it's harmful.
I think marking `retries` as private is fine too.

Reply:
Spoke offline, we'll rename the property to `playlistErrors_` to make it pseudo-private and better represent what's actually happening.